full-stack-practice/client/src/Catalog.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { Product } from '../../server/server.ts';
```

Comment:
Do not import files from your server package in your client code. They should be kept separate.

That means you have to write the Product class twice, though. There are ways to avoid that if you were working on a larger project, but for now, it's best to just have two copies of the Product class.

The same applies to ClientError below. Don't use that class in the client.